orderapi/src/main/java/fi/hel/verkkokauppa/order/listeners/DeadLetterQueueListener.java

```diff
@@ -0,0 +1,137 @@
+package fi.hel.verkkokauppa.order.listeners;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import fi.hel.verkkokauppa.common.configuration.QueueConfigurations;
+import fi.hel.verkkokauppa.common.events.EventType;
+import fi.hel.verkkokauppa.common.events.message.PaymentFailedMessage;
+import fi.hel.verkkokauppa.common.events.message.PaymentMessage;
+import fi.hel.verkkokauppa.common.queue.service.SendNotificationService;
+import fi.hel.verkkokauppa.common.rest.RestServiceClient;
+import lombok.extern.slf4j.Slf4j;
+
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.jms.annotation.JmsListener;
+import org.springframework.stereotype.Component;
+
+import javax.jms.JMSException;
+import javax.jms.TextMessage;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+
+@Component
+@Slf4j
+public class DeadLetterQueueListener {
+
+    private static final String EMAIL_TEMPLATE_PATH = "email/template_email_dlq_alert.html";
+
+    @Value("${payment.failed.notification.email:donotreply.checkout@hel.fi}")
+    private String paymentFailedNotificationEmail;
+
+    @Value("${message.service.url}")
+    private String messageServiceUrl;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @Autowired
+    private SendNotificationService sendNotificationService;
+
+    @Autowired
+    private RestServiceClient restServiceClient;
+
+    @Autowired
+    private QueueConfigurations queueConfigurations;
+
+    @JmsListener(destination = "${queue.dlq:DLQ}")
+    public void consumeMessage(TextMessage textMessage) throws Exception {
+        try {
+            PaymentMessage message = getPaymentMessageFromTextMessage(textMessage);
+
+            logMessageData(message);
+
+            if (message.getEventType().equals("PAYMENT_PAID")) {
+                paymentFailedAction(message);
+            }
+        } catch(JsonProcessingException jsonProcessingException) {
+            log.debug(jsonProcessingException.getMessage());
+            log.info("Failed to convert queue message to PaymentMessage type.");
+        }
+    }
+
+    /*
+     * This listener is only used in local development,
+     * because the default DLQ in ActiveMQ is named "ActiveMQ.DLQ" instead of "DLQ".
+     */
+    @Profile("local")
+    @JmsListener(destination = "${queue.dlq:ActiveMQ.DLQ}")
```

Comment:
Is there any delay added to this listener if there is some other than PaymentMessage:s? Then this causes a lot of errors in a loop